### PR TITLE
[25.x] backports

### PR DESCRIPTION
**Pull request by achow101**
Backport:

* #29510 
* #29412 
* #29524

### CODE REVIEW

**Review by josibake — COMMENTED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/29531/commits/f93be0103fe9cb92a376848fc534233b48977918

Verified that the correct commits are referenced in the commit messages here and that the content matches

**Comment by luke-jr** [comment]
Suggest resetting to https://github.com/bitcoin/bitcoin/compare/master...luke-jr:bitcoin:fix_reservedest_failure_pr29510-24 (merges into 24.x, 25.x, and 26.x) for cleaner backports

**Comment by achow101** [comment]
Pushed the final changes for 25.2rc2

**Comment by fanquake** [comment]
The win64 CI can be ignored, and the TSAN failure is an upstream issue, but the tests are segfaulting in the ASAN job:
```bash
Running tests: amount_tests from test/amount_tests.cpp
make[4]: Entering directory '/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src'
/bin/bash: line 2: 26108 Segmentation fault      (core dumped) test/test_bitcoin --catch_system_errors=no -l test_suite -t "$( cat test/amount_tests.cpp | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1 )" -- DEBUG_LOG_OUT > "$TEST_LOGFILE" 2>&1
make[3]: *** [Makefile:21916: test/amount_tests.cpp.test] Error 1
make[3]: *** Waiting for unfinished jobs....
```

**Comment by achow101** [comment]
> but the tests are segfaulting in the ASAN job:

Can't reproduce locally to debug.

**Comment by fanquake** [comment]
You can run the ASAN CI job locally, it recreates there.

**Comment by achow101** [comment]
```
Program received signal SIGSEGV, Segmentation fault.
0x0000618e63bd4ce1 in __sanitizer::internal_mmap(void*, unsigned long, int, int, int, unsigned long long) ()
(gdb) bt
#0  0x0000618e63bd4ce1 in __sanitizer::internal_mmap(void*, unsigned long, int, int, int, unsigned long long) ()
#1  0x0000618e63bd65da in __sanitizer::MmapNamed(void*, unsigned long, int, int, char const*) ()
#2  0x0000618e63be0398 in __sanitizer::ReservedAddressRange::Init(unsigned long, char const*, unsigned long) ()
#3  0x0000618e63b3d07a in __sanitizer::SizeClassAllocator64<__asan::AP64<__sanitizer::LocalAddressSpaceView> >::Init(int, unsigned long) ()
#4  0x0000618e63b3a37d in __asan::Allocator::InitLinkerInitialized(__asan::AllocatorOptions const&) ()
#5  0x0000618e63bc4f44 in __asan::AsanInitInternal() ()
#6  0x00007434223fd5be in _dl_init (main_map=0x7434224322e0, argc=3, argv=0x7ffe15237df8, env=0x7ffe15237e18)
    at ./elf/dl-init.c:102
#7  0x00007434224172ca in _dl_start_user () from /lib64/ld-linux-x86-64.so.2
#8  0x0000000000000003 in ?? ()
#9  0x00007ffe15239302 in ?? ()
#10 0x00007ffe15239346 in ?? ()
#11 0x00007ffe15239349 in ?? ()
#12 0x0000000000000000 in ?? ()
(gdb) 
```

Doesn't look like this is an us problem.

Also, it does not segfault consistently.

**Comment by achow101** [comment]
In my testing, it seems that the ASAN task intermittently segfaults on current 25.x. I'm unable to figure out which PR actually fixes the problem. If we're okay with ignoring CI that fails because of upstream issues, then I'm fine with ignoring this failure. But if someone wants to open a backport PR that fixes it, I'll be happy to review that.

File: doc/release-notes.md [hunk 1]
```diff
@@ -51,6 +51,12 @@ Notable changes
 ### Wallet
 
 - #29176 wallet: Fix use-after-free in WalletBatch::EraseRecords
+- #29510 wallet: `getrawchangeaddress` and `getnewaddress` failures should not affect keypools for descriptor wallets
+
+### P2P and network changes
+
+- #29412 p2p: Don't process mutated blocks
+- #29524 p2p: Don't consider blocks mutated if they don't connect to known prev block
```
Comment (glozow) [inline doc/release-notes.md:59]: Need to update the Credits section

**Comment by glozow** [comment]
backport commits lgtm

File: doc/release-notes.md (hunk 1 above)
Comment (achow101) [reply doc/release-notes.md:59]: Done

**Review by glozow — COMMENTED** [review]
utACK 27cfda1baec63ee4fb0f743576227528104fe495

**Pull request merged**